lib/src/lifecycle_module.dart

```diff
@@ -464,7 +511,15 @@ abstract class LifecycleModule extends SimpleModule with Disposable {
 
     Future pendingTransition;
     if (_transition != null && !_transition.isCompleted) {
-      pendingTransition = _transition.future;
+      pendingTransition = _transition.future
+          // Need to `catchError` before `whenComplete` to prevent errors from
+          // getting thrown here when they should be handled elsewhere
+          .catchError((_) {})
```

Comment:
When `_transition` is first created, it has a `.catchError()` attached to handle it.

However, I want to add an additional `.whenComplete()` to this same future. If I didn't have the `.catchError`, it would simultaneously be caught by the other `.catchError()` and also throw in this code. I don't care about the error in `.catchError()` handler here because it's already handled. I just have it so I can call `.whenComplete()` without this thread throwing an uncaught exception.

Reply:
I'm still a little confused... Where is the other `catchError`?

Also, if you're using `whenComplete`, you shouldn't have to add a `catchError`, since it'll get called even if future completes with an error.

I'm just not sure why `catchError` is needed here when it wasn't before these changes.

Reply:
The other catchError is in the previous transition's `await _transition`

Yes, `whenComplete` will get called, but the Future still throws synchronously on line 514.

Reply:
Sorry, I'm a little confused... Could you point out exactly where that is?

If there is a catchError there, can we instead not return the result of `catchError` to ensure it's handled by the other `catchError` and not swallowed here? 

```dart
pendingTransition = _transition.future;
_transition.future.catchError((_) {}).whenComplete(() {
  _activeSpan = _startTransitionSpan('suspend_module');
});
```

Or, if something else is calling `catchError` on the returned future and not `_transition.future`, we could also use the result of `whenComplete` without a `catchError`.

```dart
pendingTransition = _transition.future.whenComplete(() {
  _activeSpan = _startTransitionSpan('suspend_module');
});
```